### PR TITLE
Develop kodi21 patch set 2

### PR DESCRIPTION
**Pull request by bowlingbeeg**
This is the last big batch of changes I have from the ZidooPlexMod branch.

### CODE REVIEW

**Comment by pannal** [comment]
Oh wow, that's a lot of changes. 

Can you give me some details on:
* `Change watched status to also look at resume offset`
* `Fix sorting/filtering issues`

Thanks!

**Comment by bowlingbeeg** [comment]
For the watched status change that was so that once you start watching something(aka there is a viewOffset) then the little triangle to indicate the unwatched status will go away.

The sorting/filtering changes involved setting the default "titleSort" option for all of the different section types and filters including the episodes/albums.  In the current code they're using different default sorting types so this cleans that up to match what the plex server uses for the default sorting options.  I kept the other sorting option, it's just not the default anymore. I removed some of the sorting options that didn't work as well as tried to clean up the code to make it more obvious which sorting options were available for each section/filter.

**Comment by pannal** [comment]
Thanks for the explanations. Could you reply to my review question above, please?

Also, I've noticed [this in your fork](https://github.com/bowlingbeeg/plex-for-kodi/commit/8f83e5d3b9073b6485bf2f31d7feab2261daa2bb) - is that stable, could you explain the change, and would it make sense to merge, as well?

**Review by pannal — APPROVED** [review]

File: lib/util.py [hunk 1]
```diff
@@ -137,7 +137,6 @@ class AdvancedSettings(object):
         ("background_art_blur_amount", 0),
         ("background_art_opacity_amount", 40),
         ("screensaver_quiz", False),
-        ("intro_skip_early", False),
```
[the remark this answers is not in the document]
Comment (bowlingbeeg) [reply lib/util.py:140]: I removed it because it's a duplicate of line 146

**Comment by bowlingbeeg** [comment]
The changes to the updateOnDeckHubs seems to be stable.  It basically is going to update every hub that the user is actually using instead of just the hubs that have the 'do_updates' setting.  Maybe it's too heavy handed for some users with lots of sections but I didn't like how things weren't updated in the hubs after watching something so this fixed that.

**Comment by pannal** [comment]
Ah, so "speed up hub refresh" pertains to the watch state, not the actual loading time of the hub view? Thanks for clarification!

**Comment by pannal** [comment]
Hey, is there a reason why you've removed the ChunkModes in library.py?

**Comment by bowlingbeeg** [comment]
> Hey, is there a reason why you've removed the [ChunkModes in library.py](https://github.com/pannal/plex-for-kodi/pull/50/commits/1f530e84ee043e5349dcd911eacfe8bd5f5127bf)?

If I remember correctly it was because it wasn't working properly so I just removed it.

**Pull request merged**